File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.go
func bubbleFromCenter(s string, l int, r int) string {
  for l > -1 && r < len(s) && s[l] == s[r] {
    l--;
    r++;
  }
  return s[l + 1:r]
}

func longestPalindrome(s string) string {
  ans := ""
  for i := range s {
    a := bubbleFromCenter(s, i, i)
    if len(a) > len(ans) {
      ans = a
    }
    b := bubbleFromCenter(s, i, i + 1)
    if len(b) > len(ans) {
      ans = b
    }
  }
  return ans   
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
  def bubbleFromCenter(self, s: str, l: int, r: int) -> str:
    while l > -1 and r < len(s) and s[l] == s[r]:
      l -= 1
      r += 1
    return s[l + 1: r]

  def longestPalindrome(self, s: str) -> str:
    ans = ""
    for i in range(len(s)):
      a = self.bubbleFromCenter(s, i, i)
      if len(a) > len(ans): ans = a
      b = self.bubbleFromCenter(s, i, i + 1)
      if len(b) > len(ans): ans = b
    return ans   